src/endpoint/s3/s3_rest.js

```diff
@@ -121,6 +121,8 @@ async function handle_request(req, res) {
     }
     req.op_name = op_name;
 
+    http_utils.validate_bucket_name_object_key_length(req);
```

Comment:
it seems more relevant in my opinion that this validation will be part of get_bucket_and_key()